traits/trait_types.py

```diff
@@ -1496,6 +1500,12 @@ def validate(self, object, name, value):
 
         Note: The 'fast validator' version performs this check in C.
         """
+        if fspath is not None:
```

Comment:
It may be worth a comment explaining the need for the `if`, and indicating that the test can be removed once we drop Python 3.5 support. Alternatively, such a comment could go up at the top of the file, where the `fspath` import is attempted.